### PR TITLE
Fix `CursorIT.test_fetching_from_cursor_positioned_at_end_returns_empty_result`

### PR DESCRIPTION
**Pull request by matriv**
```
REPRODUCE WITH: ./gradlew :server:test --tests "io.crate.integrationtests.CursorITest.test_fetching_from_cursor_positioned_at_end_returns_empty_result" -Dtests.seed=CF1B3998759842D0 -Dtests.locale=mas-Latn-KE -Dtests.timezone=Canada/Atlantic

CursorITest > test_fetching_from_cursor_positioned_at_end_returns_empty_result FAILED
    org.opentest4j.AssertionFailedError at CursorITest.java:100
org.opentest4j.AssertionFailedError:
Expecting value to be false but was true
	at __randomizedtesting.SeedInfo.seed([CF1B3998759842D0:95666F12FFDBF0DB]:0)
```

See commit messages

### CODE REVIEW

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
irrelevant `io.crate.integrationtests.LogicalReplicationITest.test_write_to_subscribed_empty_partitioned_table_is_forbidden` failure

**Comment by matriv** [comment]
retest this please

File: server/src/main/java/io/crate/action/sql/Cursor.java [hunk 1]
```diff
@@ -235,6 +231,16 @@ private void resetCursorToMaxBufferedRowsPlus1() {
         }
     }
 
+    private int newCursorPosition(int count) {
+        int newCursorPosition = cursorPosition + count;
+        // HD 2-12 Overflow iff both arguments have the opposite sign of the result
+        if (((cursorPosition ^ newCursorPosition) & (count ^ newCursorPosition)) < 0) {
```
Comment (BaurzhanSakhariev) [inline server/src/main/java/io/crate/action/sql/Cursor.java:237]: minor: maybe we can mention in the code comments. that it's taken from `Math.add`.

**Comment by BaurzhanSakhariev** [comment]
> it maybe not guranteed that integer addition
overflow, results into a negative int

looks like its not guaranteed indeed: https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.18

>If an integer multiplication overflows, then the result is the low-order bits of the mathematical product as represented in some sufficiently large two's-complement format. As a result, if overflow occurs, then the sign of the result may not be the same as the sign of the mathematical product of the two operand values.

the sign of the result **may** not be  - I understand this "may" as undefined behavior by overflowing, similar to C. Probably that's why they introduced `Math.addExact`.

**Comment by matriv** [comment]
> looks like its not guaranteed indeed: https://docs.oracle.com/javase/specs/jls/se7/html/jls-15.html#jls-15.18
> 
> > If an integer multiplication overflows, then the result is the low-order bits of the mathematical product as represented in some sufficiently large two's-complement format. As a result, if overflow occurs, then the sign of the result may not be the same as the sign of the mathematical product of the two operand values.

This is for multiplication though, for addition (`15.18.2`)
```
If an integer addition overflows, then the result is the low-order bits of the mathematical sum as represented in some sufficiently large two's-complement format. If overflow occurs, then the sign of the result is not the same as the sign of the mathematical sum of the two operand values. 
...
...

In the remaining cases, where neither an infinity, nor a zero, nor NaN is involved, and the operands have the same sign or have different magnitudes, the exact mathematical sum is computed.

If the magnitude of the sum is too large to represent, we say the operation overflows; the result is then an infinity of appropriate sign.

Otherwise, the sum is rounded to the nearest representable value using the round to nearest rounding policy ([§15.4](https://docs.oracle.com/javase/specs/jls/se19/html/jls-15.html#jls-15.4)). The Java programming language requires support of gradual underflow.
```

but since `Math.addExact()` is doing that bitwise check, we should prefer it to the `(long)` + `> Integer.MAX_VALUE` approach.

**Comment by matriv** [comment]
@mfussenegger Please check: https://github.com/crate/crate/pull/13557/commits/001c22556634c495bd3b248eaeefa014c016d5e5 (currently I added iterations).

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
@mfussenegger Could you please check now?

**Review by mfussenegger — APPROVED** [review]

**Pull request merged**